spec/rspec/core/configuration_spec.rb

```diff
@@ -803,6 +796,42 @@ def specify_consistent_ordering_of_files_to_run
       end
     end
 
+    context "with an example id" do
+      it "assigns the file and id as an ids filter" do
+        assign_files_or_directories_to_run "./path/to/a_spec.rb[1:2]"
+        expect(inclusion_filter).to eq(:ids => { "./path/to/a_spec.rb" => ["1:2"] })
```

Comment:
The filtering implementation compares the values in the array against `example.metadata[:scoped_id]`, and `metadata[:scoped_id]` is a string in the form `x1:x2:x3:...:xn`.  It simplifies the comparison logic to parse the id into this form because we can then do a simple equality comparison.

The latter form (`{ "./path/to/a_spec.rb" => ["1", "2"] }`) is valid, but it would specify the 1st and 2nd top-level groups defined in `path/to/a_spec.rb`, which is quite different from `["1:2"]` which is the 2nd example or group in the first top-level group.